src/unix/linux-core.c

```diff
@@ -1015,14 +1025,263 @@ uint64_t uv_get_total_memory(void) {
 }
 
 
-static uint64_t uv__read_cgroups_uint64(const char* cgroup, const char* param) {
-  char filename[256];
+/*
+ * Holds information about a given cgroups subsystem.
+ */
+typedef struct {
+  /* cgroups version, one of CGROUPS_VERSION_*. */
+  uint8_t cgroups_version;
+  /* Path in which param files are found for a subsystem. */
+  char* path;
+} uv__cgroups_subsystem_info_t;
+
+
+/*
+ * Given a `separator`-delimited string `haystack`, return 1 if `needle` exactly
+ * matches at least one of the elements in that sequence.
+ */
+static int uv__find_in_delimited_string(const char* haystack, const char* needle, const char* separator) {
+  char* haystack_mutable;
+  size_t haystack_size;
+  char* candidate;
+  char* haystack_ptr;
+  haystack_size = strlen(haystack) + 1;
+  if (needle == NULL || strlen(needle) >= haystack_size)
+    return 0;
+  haystack_mutable = uv__malloc(haystack_size);
```

Comment:
maybe this can be replaced with `uv__strndup` ? https://github.com/libuv/libuv/blob/89a027d862c9b84a20b0316415be169ac25dbf01/src/uv-common.c#L63

Reply:
Done